pkg/build/buildkit/buildx.go

```diff
@@ -59,89 +65,154 @@ func (l unstructuredLogger) Write(p []byte) (n int, err error) {
 	)
 	if err != nil {
-		return log.Error(err)
+		return span.Error(err)
 	}
 	nodes, err := bldr.LoadNodes(ctx, false)
 	if err != nil {
-		return log.Error(err)
+		return span.Error(err)
 	}
 
-	session := []session.Attachable{authprovider.NewDockerAuthProvider(dockerconfig.LoadDefaultConfigFile(b.Err))}
+	currentSession := []session.Attachable{authprovider.NewDockerAuthProvider(dockerconfig.LoadDefaultConfigFile(b.Err))}
 	ssh, err := buildflags.ParseSSHSpecs(opts.SSH)
 	if err != nil {
-		return fmt.Errorf("error parsing the --ssh flags: %w", err)
+		return span.Errorf("error parsing the --ssh flags: %w", err)
 	}
-	session = append(session, ssh)
+	currentSession = append(currentSession, ssh)
 
 	secrets, err := buildflags.ParseSecretSpecs(opts.Secrets)
 	if err != nil {
-		return fmt.Errorf("error parsing the --secret flags: %w", err)
+		return span.Errorf("error parsing the --secret flags: %w", err)
 	}
-	session = append(session, secrets)
+	currentSession = append(currentSession, secrets)
 
-	args := make(map[string]string, len(opts.BuildArgs)+1)
-	parseBuildArgs(opts.BuildArgs, args)
-	args["BUNDLE_DIR"] = build.BUNDLE_DIR
-
-	convertedCustomInput, err := flattenMap(manifest.Custom)
+	args, err := b.determineBuildArgs(ctx, manifest, opts)
 	if err != nil {
-		return log.Error(err)
-	}
-
-	for k, v := range convertedCustomInput {
-		args[strings.ToUpper(strings.Replace(k, ".", "_", -1))] = v
+		return err
 	}
+	span.SetAttributes(tracing.ObjectAttribute("build-args", args))
 
 	buildxOpts := map[string]buildx.Options{
 		"default": {
 			Tags: []string{manifest.Image},
 			Inputs: buildx.Inputs{
 				ContextPath:    b.Getwd(),
-				DockerfilePath: filepath.Join(b.Getwd(), build.DOCKER_FILE),
+				DockerfilePath: b.getDockerfilePath(),
 				InStream:       b.In,
 			},
 			BuildArgs: args,
-			Session:   session,
+			Session:   currentSession,
 			NoCache:   opts.NoCache,
 		},
 	}
 
 	mode := progress.PrinterModeAuto // Auto writes to stderr regardless of what you pass in
-	out := unstructuredLogger{log}
+	out := unstructuredLogger{span}
 	printer, err := progress.NewPrinter(ctx, out, os.Stderr, mode)
 	if err != nil {
-		return log.Error(err)
+		return span.Error(err)
 	}
 
 	_, buildErr := buildx.Build(ctx, nodes, buildxOpts, dockerutil.NewClient(cli), confutil.ConfigDir(cli), printer)
 	printErr := printer.Wait()
 
 	if buildErr == nil && printErr != nil {
-		return log.Error(fmt.Errorf("error with docker printer: %w", printErr))
+		return span.Errorf("error with docker printer: %w", printErr)
 	}
 
 	if buildErr != nil {
-		return log.Error(fmt.Errorf("error building docker image: %w", buildErr))
+		return span.Errorf("error building docker image: %w", buildErr)
 	}
 
 	return nil
 }
 
+func (b *Builder) getDockerfilePath() string {
+	return filepath.Join(b.Getwd(), build.DOCKER_FILE)
+}
+
+func (b *Builder) determineBuildArgs(ctx context.Context, manifest *manifest.Manifest, opts build.BuildImageOptions) (map[string]string, error) {
```

Comment:
nit: maybe putting args on newlines would be cleaner?

```suggestion
func (b *Builder) determineBuildArgs(
    ctx context.Context,
    manifest *manifest.Manifest,
    opts build.BuildImageOptions
) (map[string]string, error) {
```